scripts/linters/html_linter.py

```diff
@@ -209,6 +214,41 @@ def handle_data(self, data):
             elif data_line.startswith(ending_block):
                 self.indentation_level -= 1
 
+    def _check_space_between_attributes_and_values(
+            self, tag, attr, value, rendered_text, value_in_quotes):
+        """Checks if there are any spaces between attributes and their values.
+
+        Args:
+            tag: str. The tag name of the HTML line.
+            attr: str. The attribute name in the tag.
+            value: str. The value of the attribute.
+            rendered_text: str. The rendered text of the tag.
+            value_in_quotes: bool. Whether the given attribute value
+                is in double quotes.
+        """
+
+        attr_val_structure = '{}="{}"' if value_in_quotes else '{}={}'
+        line_number, _ = self.getpos()
+        if rendered_text.find(attr) < 0:
+            # Attribute names are case insensitive,
+            # So attr name might not match the
+            # actual attr name in tag.
+
+            attr_pos = rendered_text.lower().find(attr)
+            attr_name = rendered_text[attr_pos:attr_pos + len(attr)]
+            expected_attr_assignment = attr_val_structure.format(
+                attr_name, value)
+        else:
+            expected_attr_assignment = attr_val_structure.format(
+                attr, value)
+        if rendered_text.find(expected_attr_assignment) < 0:
```

Comment:
should we use `expected_attr_assignment in render_text`?

Reply:
it should be not in, but yeah, Done